units/cache_historical_test_data.go

```diff
@@ -0,0 +1,289 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+
+	"github.com/evergreen-ci/evergreen/model/stats"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	cacheHistoricalTestDataName = "cache-historical-test-data"
+	maxSyncDuration             = time.Hour * 24 * 7 // one week
+)
+
+func init() {
+	registry.AddJobType(cacheHistoricalTestDataName,
+		func() amboy.Job { return makeCacheHistoricalTestDataJob() })
+}
+
+type cacheHistoricalTestDataJob struct {
+	ProjectId string `bson:"project_id" json:"project_id" yaml:"project_id"`
+	job.Base  `bson:"job_base" json:"job_base" yaml:"job_base"`
+}
+
+type dailyStatsRollup map[time.Time]map[string][]string
+
+type generateStatsFn func(string, string, time.Time, []string, time.Time) error
+type generateFunctions struct {
+	HourlyFns map[string]generateStatsFn
+	DailyFns  map[string]generateStatsFn
+}
+
+func NewCacheHistoricalTestDataJob(projectId string, id string) amboy.Job {
+	j := makeCacheHistoricalTestDataJob()
+	j.ProjectId = projectId
+	j.SetID(fmt.Sprintf("%s-%s-%s", cacheHistoricalTestDataName, projectId, id))
+	return j
+}
+
+func makeCacheHistoricalTestDataJob() *cacheHistoricalTestDataJob {
+	j := &cacheHistoricalTestDataJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    cacheHistoricalTestDataName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func (j *cacheHistoricalTestDataJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	projectId := j.ProjectId
+
+	grip.Info(message.Fields{
+		"job_id":     j.ID(),
+		"project_id": projectId,
+		"message":    "starting job",
+	})
```

Comment:
the job system logs start/stop for jobs so we can omit this.

Reply:
removed.